extensions/amp-sidebar/1.0/toolbar.js

```diff
@@ -103,36 +100,19 @@ export class Toolbar {
     this.toolbarClone_ = this.toolbarDOMElement_.cloneNode(true);
     const targetId = this.toolbarDOMElement_.getAttribute('target');
     // Set the target element to the toolbar clone if it exists.
-    const targetElement =
-    this.win_.document.getElementById(`${targetId}_toolbar`) ||
-    this.win_.document.getElementById(targetId);
-    this.targetElement_ = targetElement || this.createTargetElement_();
-    this.targetElement_.appendChild(this.toolbarClone_);
-    toggle(this.targetElement_, false);
-
-    // Check if the target element was created by us, or already inserted by the user
-    if (!this.targetElement_.parentElement) {
+    const targetElement = this.win_.document.getElementById(targetId);
```

Comment:
Neither `build` or `layout` phase (or any other) would work. There's simply no way to relate a phase of any element to another element - they are completely disconnected and async. You probably should use `ampdoc.whenReady().then({getElementById()})` or such. This would allow you to search DOM once you know it's fully parsed. I don't think you need to do this faster than that.